### PR TITLE
HAR file creation and running with Phantomjs

### PR DESCRIPTION
**Pull request by sstaedt**

### CODE REVIEW

**Comment by scott2449** [comment]
:+1: looks great I'll review further when I get back.  Code reviews are really hard while sitting in Washington square park on a sunny afternoon :smile:

**Comment by scott2449** [comment]
As discussed we need tests and coverage that is at least no less then before 95%ish. Also lets pull as much of the HAR code as we can into a separate file, this will make for cleaner implementation and easier testing. Otherwise LOVE this feature!

**Comment by scott2449** [comment]
@sstaedt eta? I'd like to push out 1.1.0 next week. If you are busy I can work on this and squash my commits into yours so you still get all the cred :smiley_cat:

**Comment by sstaedt** [comment]
Let me see what I can get done this weekend. I should have time, with a storm moving in and all 😄.

**Comment by sstaedt** [comment]
OK. Here is the refactor. Hope this helps.

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "gulp-istanbul": "^0.3.0",
     "gulp-mocha": "^1.1.1",
     "gulp-coveralls": "^0.1.3",
-    "proxyquire": "^1.0.1"
+    "proxyquire": "^1.0.1",
+    "child-proc": "^0.0.1"
```
Comment (scott2449) [inline package.json:28]: what's this dep for? I don't see it in the test or code.

**Comment by scott2449** [comment]
@sstaedt  aside from my question about the dep it looks fantastic now!

File: lib/logManager.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+logManager.prototype.writeFile = function (directoryPath, filename, message) {
+  var fs = require('fs');
+
+  if(!fs.existsSync(directoryPath)){
+    fs.mkdirSync(directoryPath, 0766, function(err){
+      if(err){
+        console.error(err);
+        response.send("ERROR! Can't make the directory for har files! \n");    // echo the result back
```
Comment (scott2449) [inline lib/logManager.js:49]: also where is this response var coming from?

File: package.json (hunk 1 above)
Comment (sstaedt) [inline package.json:28]: This dep is for the last test for logManager. I wanted to remove the dir 'harfiles' recursively before executing to make sure it creates a dir when it doesn't already exist. That part can be removed if we always run the tests with a clean slate.

File: lib/logManager.js (hunk 2 above)
Comment (sstaedt) [inline lib/logManager.js:49]: Meant to remove that line of code. Oops.

**Comment by scott2449** [comment]
@sstaedt thanks for the answers I'll take care of the rest of the cleanup after I merge the PR. I am excited for 1.1.0!!

**Pull request merged**